torch/_subclasses/fake_tensor.py

```diff
@@ -318,7 +312,8 @@ def non_kwarg_to(fake_mode, func, *args, **kwargs):
     new_kwargs["device"] = torch.device("meta")
     inp = new_kwargs.pop("input")
     r = func(inp, **new_kwargs)
-    return fake_mode.fake_tensor_converter(fake_mode, r, out_device)
+    # TODO: I think this does the wrong thing if r is inp
```

Comment:
these are guaranteed copies, i thought. this is post-composite implicit. no-op aten::to shouldn't invoke `__torch_dispatch__` right ?

Reply:
If my comment is correct can you remove `# TODO: I think this does the wrong thing if r is inp` comment before landing

Reply:
You registered this implementation to `aten.to_device` which is a composite

```
- func: to.device(Tensor(a) self, Device device, ScalarType dtype, bool non_blocking=False, bool copy=False, MemoryFormat? memory_format=None) -> Tensor(a)
  variants: method
  device_check: NoCheck
  device_guard: False
```

as its alias annotations say, it can return an alias of the input tensor.

Reply:
Indeed, `.to()` can return `self` if nothing is being done.
Should this be:
```python
if r is inp:
  return r
else:
  return fake_mode....
```

Reply:
Got it. I didn't think composites could even be hit in `__torch__dispatch__`, so I guess I didn't check alias carefully. Is that not a bug ?

Reply:
Well yeah, my suspicion is that override is dead. Need to test.

Reply:
Well, if the converter is a no-op for already fake Tensor, that should be fine.